### ISSUE TITLE
Svelte 5: Parsing Error with Snippets

### ISSUE THREAD

**Issue by abdel-17** [sveltejs/eslint-plugin-svelte#686]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.

### What version of ESLint are you using?

8.56.0

### What version of `eslint-plugin-svelte` are you using?

2.63.0-next.5

### What did you do?

ESlint fails to parse snippets whose arguments are function calls.
```html
<script lang="ts">
	import type { Snippet } from "svelte";

	let { foo } = $props<{ foo: Snippet<[string]> }>();

	function bar() {
		return "baz";
	}
</script>

{@render foo(bar())}
```

### What did you expect to happen?

Parsing should not fail.

https://github.com/sveltejs/eslint-plugin-svelte/assets/88583085/832869b9-2b9d-4a1f-b598-2972416e82c3

### What actually happened?

```bash
  13:11  error  Parsing error: ')' expected

✖ 1 problem (1 error, 0 warnings)
```

### Link to **GitHub Repo** with Minimal Reproducible Example

https://github.com/abdel-17/svelte-eslint-bug-reproduction
